### DISCUSSION TITLE
Ignoring a warning for a multi-line statement

### DISCUSSION THREAD

**Discussion by karolyi** [microsoft/pyright#8349] [Q&A]
Hello,

I'm using 80 columns as a max width, and I'm running into problems where I break long lines and the pyright ignore statement can't be used at the first line, or on any other intermittent line that is not the last one. Example:

```python
class X(object):
    def stuff(self):
        self.super_long_declaration_name_that_doesnt_fit_in_one_line: int = \
            5 * 5  # pyright: ignore[reportAttributeAccessIssue]
```

This ignore statement can't be used on the `self.super_long_declaration_name_that_doesnt_fit_in_one_line` declaration, although it should be used on that one, and not on the second line (which has no errors).

Not to mention that the specific pyright line ignore statements themselves can be way too long, there is no way to tell it to ignore the entire statement, instead of just the line it's put on.

I'm looking for a way to have this actually working.  Using a `# pyright:ignore` statement after the backslash is not an option as it makes the code syntactically wrong.

Please don't tell me I have no other way to solve this other than to enable lines with arbitrary character count.

**Comment by erictraut** [comment]
Here are some options:
1. Ignore the 80-column limit in this instance and place the `# pyright: ignore` at the end of the line. The python interpreter obviously doesn't care how many columns your code is, so this is a self-imposed limit, and you may want to be a bit more lax about it. Modern code auto-formatters like black or ruff allow end-of-line comments to go past the defined limit specifically to support use cases like this. If, for some reason, you want to be really strict about not exceeding 80 characters, your options are rather limited.
2. Find some other way to avoid the diagnostic so you don't need to suppress it. I can't offer any more specific advice here because I don't have enough context about your code.
3. Reduce the length of `super_long_declaration_name_that_doesnt_fit_in_one_line` to something that doesn't consume most of your 80-character budget.
4. Use a `# pyright: reportAttributeAccessIssue=false` at the top of the file to suppress these errors completely.
5. Disable `reportAttributeAccessIssue` globally in your project using the config file.

**Reply by karolyi** [reply]
Hey Eric,

thanks for the insight.

I managed to work it around. The actual code is different and you might have noticed that the ignored pyright errors are not valid in this context.

For some reason, when I put the code differently, the actual error starts to appear on the function/method call, and not on the variable. Weird, but the moment it appears there, I am able to break that line into more parts and add an ignore statement. My actual example here is:

```python
class PaymentPkDbAggregator(object):
    'Creating and handling aggregated queries.'
    _INSERT_SQL = 'INSERT INTO `{table}` (`{unique_id}`, `{payment_fk}`)'
    _payment_fk_field: ForeignKey[Payment, Payment] = \
        IncomeStatAggregator._meta.get_field(
            'payment')  # pyright: ignore[reportAssignmentType]
    _unique_id_field: CharField[str, str] = \
        IncomeStatAggregator._meta.get_field(
            'unique_id')  # pyright: ignore[reportAssignmentType]
    _table_name = IncomeStatAggregator._meta.db_table

```

This way I don't get the errors for the lines.

The original format was:

```python
    def __init__(self, request: HttpRequest):
        self.request = request
        self._table_name = IncomeStatAggregator._meta.db_table
        self._payment_fk_field: ForeignKey[Payment, Payment] = \
            IncomeStatAggregator._meta.get_field('payment')
        self._unique_id_field: CharField[str, str] = \
            IncomeStatAggregator._meta.get_field('unique_id')
        self.unique_ids = list()
```

In this latter fashion, the errors appeared on the variable names:

![image](https://github.com/microsoft/pyright/assets/987055/298e4eb0-eb68-41bc-a4a3-30283a753155)

Not sure this helps in outlining the problem better, or finding any bugs in pyright, but here it is. This is code used in a Django project.
